plugins/module_utils/dnac.py

```diff
@@ -362,9 +362,10 @@ def check_task_response_status(self, response, validation_string, api_name, data
         while True:
             end_time = time.time()
             if (end_time - start_time) >= self.max_timeout:
-                self.log("Max timeout of {0} sec has reached for the execution id '{1}'. "
-                         "Exiting the loop due to unexpected API '{2}' status."
-                         .format(self.max_timeout, task_id, api_name), "WARNING")
+                self.msg = "Max timeout of {0} sec has reached for the task id '{1}'. ".format(self.max_timeout, task_id) + \
```

Comment:
self.msg = "Max timeout of {0} sec has been reached for the task id '{1}'. ".format(self.max_timeout, task_id) + \
               "Exiting the loop due to unexpected API '{0}' status.".format(api_name)

Do we need to Correct the format placeholder in the second formatted string from {2} to {0} to correctly insert api_name? Can you please check? 

I recommend to write like below:
```
   self.msg = "Max timeout of {max_timeout} second(s) has been reached for the task id '{task_id}'. ".format(
        max_timeout=self.max_timeout, task_id=task_id) + \
               "Exiting the loop due to unexpected API '{api_name}' status.".format(api_name=api_name)
    self.log(self.msg, "WARNING")
```

Or better would be:
```
   self.msg = ("Max timeout of {max_timeout} second(s) has been reached for the task id '{task_id}'. "
                "Exiting the loop due to unexpected API '{api_name}' status.").format(
                    max_timeout=self.max_timeout, task_id=task_id, api_name=api_name)
```
    self.log(self.msg, "WARNING")